Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/DocumentObj/SearchIshDocumentObj.cs

```diff
@@ -0,0 +1,357 @@
+                        //debug//   <ishobjectfilters><ishversionfilter>LatestVersion</ishversionfilter><ishtypefilter>ISHModule</ishtypefilter><ishtypefilter>ISHMasterDoc</ishtypefilter><ishlanguagefilter>en</ishlanguagefilter>
+                        //debug//   </ishobjectfilters></ishquery>";
+                        XmlDocument xmlDocument = new XmlDocument();
+                        xmlDocument.LoadXml(_xmlQuery);
+                        _xmlQuery = xmlDocument.OuterXml;
+                        WriteDebug("Validated incoming XmlQuery[" + xmlDocument.OuterXml + "]");
+                        break;
+                }
+
+                // 3. Executing
+                switch (ParameterSetName)
+                {
+                    case "SimpleQueryCountGroup":
+                    case "XmlQueryCountGroup":
+                        {
+                            // 3A. Executing the count query
+                            WriteVerbose("Executing a count query");
+                            _maxHitsToReturn = 0; // because that is what -Count does
+                            Search25ServiceReference.PerformSearchResponse performSearchResponse = IshSession.Search25.PerformSearch(new Search25ServiceReference.PerformSearchRequest(_xmlQuery, _maxHitsToReturn));
+                            WriteObject(performSearchResponse.totalHitsFound);
+                        }
+                        break;
+
+                    case "XmlQueryGroup":
+                    case "SimpleQueryGroup":
+                        {
+                            // 3B. Executing the result query
+                            WriteVerbose("Executing a result query");
+                            Search25ServiceReference.PerformSearchResponse performSearchResponse = IshSession.Search25.PerformSearch(new Search25ServiceReference.PerformSearchRequest(_xmlQuery, _maxHitsToReturn));
+                            var lngCardIds = new IshSearchResults(performSearchResponse.xmlSearchResults).LngRefs;
+
+                            if (lngCardIds.Count <= 0)
+                            {
+                                WriteWarning($"Search-IshDocumentObj returned 0 results for xmlQuery[{_xmlQuery}]");
+                            }
+                            else
+                            { 
+                                // 4. Retrieving metadata on the search result
+                                WriteDebug("Retrieving metadata on the search result");
+                                IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, new IshFields(RequestedMetadata), Enumerations.ActionMode.Read);
+
+                                //RetrieveMetadata
+                                WriteDebug("Retrieving CardIds.length[" + lngCardIds.Count + "] RequestedMetadata.length[" + requestedMetadata.ToXml().Length + "] 0/" + lngCardIds.Count);
```

Comment:
Minor observation..it is probably similar to GetIshDocumentObj, but thinking if we really need to write WriteDebug with "0/" + lngCardIds.Count
Let's say I have MaxHitsToReturn=10000 and batchSize=1000 in Search-IshDocumentObj -SimpleQuery "*" -MaxHitsToReturn 10000
Then I expect to see 10 "progress" Debug lines, not 11. The first one will be 
"DEBUG: SearchIshDocumentObj 20210405.180207.065 Retrieving CardIds.length[10000] RequestedMetadata.length[5844] 0/10000" and I am not sure if it contains useful information